staging/kos/pkg/controllers/connectionagent/network_interface.go

```diff
@@ -208,17 +218,32 @@ func (ca *ConnectionAgent) createRemoteNetworkInterface(att *netv1a1.NetworkAtta
 	err := ca.netFabric.CreateRemoteIfc(ifc.RemoteNetIfc)
 	tAfter := time.Now()
 
-	ca.fabricLatencyHistograms.With(prometheus.Labels{"op": "CreateRemoteIfc", "err": formatErrVal(err != nil)}).Observe(tAfter.Sub(tBefore).Seconds())
+	ca.fabricLatencyHistograms.With(prometheus.Labels{opLabel: "CreateRemoteIfc", errLabel: formatErrVal(err != nil)}).Observe(tAfter.Sub(tBefore).Seconds())
 	if err == nil {
+		// Record some metrics.
 		naSpecWriteTime := att.Writes.GetServerWriteTime(netv1a1.NASectionSpec).Time()
 		naImplWriteTime := att.Writes.GetServerWriteTime(netv1a1.NASectionImpl).Time()
-		if vnRelevanceTime.Before(naImplWriteTime) {
-			ca.attachmentCreateToRemoteIfcHistogram.Observe(tAfter.Sub(naSpecWriteTime).Seconds())
-			ca.localImplToRemoteIfcHistogram.Observe(tAfter.Sub(naImplWriteTime).Seconds())
+		pl := prometheus.Labels{
+			naWaitedForIPAMLabel:    att.Status.WaitedForIPAM.String(),
+			naWaitedForLocalCALabel: att.Status.WaitedForCA.String(),
+		}
+		var (
+			attCreateToRemIfcDt float64
+			localImplToRemIfcDt float64
+		)
+		if naImplWriteTime.Before(vnRelevanceTime) {
+			pl[remoteCAStartedLastLabel] = strconv.FormatBool(ca.startTime.After(vnRelevanceTime))
+			pl[naReadyBeforeVNRelevanceLabel] = "true"
+			attCreateToRemIfcDt = tAfter.Sub(vnRelevanceTime).Seconds() + naImplWriteTime.Sub(naSpecWriteTime).Seconds()
```

Comment:
This is excluding some latency that actually happened.  I think I have a preference for not excluding any latency but rather adding an explanation of it.

Reply:
Note that you talk about the write of the local impl's status as if that is the starting gun for a remote implementation, but this time depends on KOS performance.  I think the proper starting gun is the last client write that make the remote implementation desired.

Let's think about the two common cases, using the example and names in https://github.com/MikeSpreitzer/kube-examples/blob/add-kos/staging/kos/doc/KOS-causaility.pdf .  In the common cases all the process starts precede the creations of the objects involved and no object deletes are inolved.  That leaves two cases, suggested by the horizontal positioning in that diagram: (1) the VN is relevant on node N before the remote NA is created, and (2) the VN becomes relevant to node N after the remote NA is created.  In case 1, it is meaningful to report the latency from NA creation to remote implementation.  In case 2, no number is meaningful for that latency; that is about a story that simply did not happen.  In case 2 the CA should report latency from when the VN became relevant to when the NA's remote implementation was done.